packages/sds/src/message_channel/mem_local_history.ts

```diff
@@ -11,8 +12,23 @@ import { ContentMessage, isContentMessage } from "./message.js";
  *
  * Only stores content message: `message.lamportTimestamp` and `message.content` are present.
  */
-export class MemLocalHistory {
+export class MemLocalHistory implements ILocalHistory {
   private items: ContentMessage[] = [];
+  [index: number]: ContentMessage | undefined;
+
+  public constructor() {
+    // Set up a Proxy to handle index access
+    return new Proxy(this, {
```

Comment:
I think we shouldn't do this.
Quickly checking how this entity is used - I think we should implement something else here.

Used API:
https://github.com/waku-org/js-waku/blob/18f30f509f49cef98d707f5d9d1abc5c718915d0/packages/sds/src/message_channel/message_channel.ts#L632C5-L634C7
https://github.com/waku-org/js-waku/blob/18f30f509f49cef98d707f5d9d1abc5c718915d0/packages/sds/src/message_channel/message_channel.ts#L645
https://github.com/waku-org/js-waku/blob/18f30f509f49cef98d707f5d9d1abc5c718915d0/packages/sds/src/message_channel/message_channel.ts#L465

From this I see that better way would be to implement `Map`:
- `get(messageId)`
- `has(messageId)`
- `add(message)`

Also, it seems like `BloomFilter` is often used together so maybe we can even combine them in future

Reply:
This is used to build the local history. Hence we need to keep the array ordered as per protocol specs. Hence why its an array and not a set or map.

We also want to avoid duplicates.

Happy to consider alternatives but Map isn't a valid one.

Reply:
The `Proxy` is actually just for a test and I don't think a real consumer would use it. I will try to remove it and just change the test.

Reply:
Just to be clear, I don't understand how you would implement `some` and `slice` with `Map`.